### PR TITLE
Commit overlay able to redirect to commit on Github

### PR DESCRIPTION
**Pull request by CaseyHillers**

### CODE REVIEW

File: app_flutter/test/commit_box_test.dart [hunk 1]
```diff
@@ -71,5 +72,50 @@ void main() {
       // Image.Network throws a 400 exception in tests
       tester.takeException();
     });
+
+    testWidgets('tapping redirect button should redirect to Github',
```
Comment (digiter) [inline app_flutter/test/commit_box_test.dart]: Is it possible to use a fake or mock url_launcher in this test? @jonahwilliams

Comment (jonahwilliams) [reply app_flutter/test/commit_box_test.dart]: That would be the setMockMethodCallHandler below

Comment (goderbauer) [reply app_flutter/test/commit_box_test.dart]: Isn't that what the setMockMethodCallHandler is for?

Comment (CaseyHillers) [reply app_flutter/test/commit_box_test.dart]: If it helps, I just did what the plugin does :)
https://github.com/flutter/plugins/blob/master/packages/url_launcher/test/url_launcher_test.dart

File: app_flutter/test/commit_box_test.dart [hunk 2]
```diff
@@ -71,5 +72,50 @@ void main() {
...
+        log.add(methodCall);
+      });
+
+      await tester.pumpWidget(MaterialApp(
+        home: CommitBox(
+          commit: expectedCommit,
+        ),
+      ));
+
+      tester.takeException();
```
Comment (goderbauer) [inline app_flutter/test/commit_box_test.dart]: Why are we ignoring exceptions here and below? Maybe add a comment with a quick explanation or (better:) add an expect to assert that the exception returned is actually what you expect it to be.

File: app_flutter/test/commit_box_test.dart (hunk 1 above)
Comment (digiter) [reply app_flutter/test/commit_box_test.dart]: They are not the same. Using MockMethodCallHandler is testing the implementation of url_handler, which is worse than testing the interface. With a fake url_handler, the test can be independent of the implementation.

File: app_flutter/test/commit_box_test.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_flutter/test/commit_box_test.dart]: Removed, it was left over :)

File: app_flutter/test/commit_box_test.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_flutter/test/commit_box_test.dart]: I agree on this testing the implementation too much, but this is a limitation of the url_launcher plugin and is out of scope for this project. Right now this test ensures the functionality and may need updates if url_launcher updates. Other than that, I don't have a solution that can ensure the functionality is still working.

Should I file a feature request to the plugin for better testing support @digiter ?

Comment (digiter) [reply app_flutter/test/commit_box_test.dart]: I agree this is out of scope. I just want to learn the typical approach from others.

**Pull request merged**